### ISSUE TITLE
Support Windows High Contrast mode

### ISSUE THREAD

**Issue by llemeurfr** [readium/readium-desktop#577]
The [High Contrast Mode](https://www.youtube.com/watch?v=MU1fDBSps5g) is a system mode, where users can choose background and foreground color, buttons and links, selected text. 

Support on this mode has to go through an evolution of Readium CSS.
Seems related to https://github.com/readium/readium-css/issues/6 (themes) and somewhat related to  https://github.com/readium/readium-css/issues/65 (issue about highlight).

**Comment by JayPanoz** [comment]
Related ReadiumCSS module: [ReadiumCSS-os_a11y.css](https://github.com/readium/readium-css/blob/master/css/src/modules/ReadiumCSS-os_a11y.css) (it’s more of a placeholder/WIP currently, because media queries are moving at a very quick pace)

Question though: does Electron has any ability to detect high-contrast-mode in Windows? 

Because currently, there’s `-ms-high-contrast` but it only works in Microsoft’s IE/Edge:

- [MDN](https://developer.mozilla.org/en-US/docs/Web/CSS/@media/-ms-high-contrast)
- [how-to](https://www.gwhitworth.com/blog/2017/04/how-to-use-ms-high-contrast/)

And the contrast/forced colors/transparency queries are currently being standardised in [Level 5](https://drafts.csswg.org/mediaqueries-5/#prefers-contrast)…

**Comment by JayPanoz** [comment]
Note you have to use an extension in Chrome for high-contrast: [chromium low-vision page](https://www.chromium.org/user-experience/low-vision-support)

**Comment by JayPanoz** [comment]
And related Electron issue: https://github.com/electron/electron/issues/13178

**Comment by danielweck** [comment]
Related issue: https://github.com/readium/readium-desktop/issues/83

**Comment by danielweck** [comment]
See note about SVG icons:
https://github.com/readium/readium-desktop/issues/410#issuecomment-535378748

**Comment by Scarsniik** [comment]
In the version of electron we are using, there isn't the event I need to use to detect that the high contrast mode is enable. When do you think you will migrate to electron 6 ?

**Comment by danielweck** [comment]
ASAP. I don't want to leave this for later. Let's tackle the migration early on, so we can run lots of tests.

**Comment by danielweck** [comment]
Navigator first:
https://github.com/readium/r2-navigator-js/compare/feature/electron5?expand=1

**Comment by danielweck** [comment]
Testapp second:
https://github.com/readium/r2-testapp-js/compare/feature/electron5?expand=1

**Comment by Scarsniik** [comment]
Ok, cool
Then I'll do it without the event and put it in WIP until we use the good version

**Comment by danielweck** [comment]
Also note that Electron6 has a known bug with Windows dark mode in developer mode (developer tools not compatible), so as developers, we must switch to light mode when using the dev tools.

**Comment by panaC** [comment]
https://electronjs.org/docs/api/system-preferences#event-high-contrast-color-scheme-changed-windows

**Comment by panaC** [comment]
https://www.electronjs.org/docs/tutorial/mojave-dark-mode-guide

**Comment by panaC** [comment]
what is the progress status of this issue ?

it's narrow linked with the css. It may rather be a dedicated task for the css master.

**Comment by danielweck** [comment]
Merged issue: https://github.com/edrlab/thorium-reader/issues/1726

### Description
Under Windows, if a high-contrast theme is used, some of the elements in the UI of Thorium Reader are not visible. Examples of such elements are the buttons in the top toolbar and the button with actions for a book. See attached screenshots for examples.

### Steps to reproduce
1. Enable high-contrast mode in Windows 11, by doing the following:
1.1. Open Windows Settings (Windows+I).
1.2. Go to Accessibility ➡ Contrast themes.
1.3. From the "Contrast themes" combo-box, select any dark theme (e.g. "Night Sky").
1.4. Activate the "Apply" button.
2. Launch Thorium Reader.
3. If you don't have any books added, add at least one.
4. Observe how the button with the 3 vertical dots (the one that opens the menu for a book) is not visible.
5. Open for reading one of the added books.
6. Observe how the buttons in the toolbar on the top are not visible.

### Test environment
- Operating system: Windows 11 Pro version 21H2 (build 22000.795), 64-bit, in Bulgarian with all locale settings set to "Bulgarian".
- Thorium Reader version: v2.0.0, installed from the Microsoft Store.

![2022-07-25_09-21-50](https://user-images.githubusercontent.com/4395934/180727350-e00aae04-a950-4b78-a3c8-b56019709bcd.png)
![2022-07-25_09-23-00](https://user-images.githubusercontent.com/4395934/180727354-3208961b-4fdd-4056-8126-a063f5fc0d17.png)

![2022-07-25_09-24-19](https://user-images.githubusercontent.com/4395934/180727358-b40ffac0-3db9-4b91-9921-1bed1401e664.png)
![2022-07-25_09-24-58](https://user-images.githubusercontent.com/4395934/180727364-5b82dbd9-ae4a-4470-b474-537abb45d465.png)
![2022-07-25_09-24-45](https://user-images.githubusercontent.com/4395934/180727362-c0a8c25c-15f7-47e1-9c9c-098cd30d523a.png)
![2022-07-25_09-23-46](https://user-images.githubusercontent.com/4395934/180727356-7055056f-cd6e-408f-b32b-d6b845c740c1.png)

**Comment by danielweck** [comment]
forced colours media query: https://developer.mozilla.org/en-US/docs/Web/CSS/@media/forced-colors

**Comment by danielweck** [comment]
Besides pure-CSS techniques, let's not forget Electron's own system preferences API: https://www.electronjs.org/docs/latest/api/system-preferences#systempreferencesgetcolorcolor-windows-macos
(we can also listen to colour theme change events)

**Comment by danielweck** [comment]
I don't think we will have time to implement this for Thorium 3.0 but definitely something to explore in 3.1 as a logical follow-up to the introduction of system dark/light modes and additional colour themes

**Comment by panaC** [comment]
We cannot do this in Thorium3 but instead in Thorium3.x maybe

**Comment by danielweck** [comment]
@panaC recently tested this on his Windows workstation, some things appear to work fine but we need to fully test the GUI as well as the publication document styles (including annotation highlights, bookmark margin indicators, search result highlights, TTS highlights and EPUB Media Overlays)
